sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/DigitalTwinMetadataJsonConverter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DigitalTwins.Core
+{
+    /// <summary>
+    /// JSON converter to make it easier to deserialize a <see cref="BasicDigitalTwin"/>.
+    /// </summary>
+    internal class DigitalTwinMetadataJsonConverter : JsonConverter<DigitalTwinMetadata>
+    {
+        /// <inheritdoc/>
+        public override DigitalTwinMetadata Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            if (reader.TokenType == JsonTokenType.StartObject)
+            {
+                var metadata = new DigitalTwinMetadata();
+                reader.Read(); // Advance into our object.
+
+                while (reader.TokenType != JsonTokenType.EndObject) // Until we reach the end of the object we began reading
+                {
+                    string propertyName = reader.GetString();
+                    if (reader.TokenType == JsonTokenType.PropertyName)
+                    {
+                        if (propertyName == DigitalTwinsJsonPropertyNames.MetadataModel)
+                        {
+                            reader.Read(); // advance to the string
+                            metadata.ModelId = JsonSerializer.Deserialize<string>(ref reader, options);
+                            reader.Read(); // Consume the string we deserialized
+                        }
+                        else
+                        {
+                            reader.Read(); // Should place us on StartObject
```

Comment:
since in both if/else blocks, probably move outside. Unless it's there to clarify the line comment.